Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift

```diff
@@ -79,5 +87,76 @@ extension Snapshotting where Value: SwiftUI.View, Format == UIImage {
       }
   }
 }
+
+#if compiler(>=5.7)
+@available(iOS 16.0, tvOS 16.0, *)
+extension Snapshotting where Value: SwiftUI.View, Format == UIImage {
+  /// A snapshot strategy for comparing SwiftUI Views based on pixel equality using iOS 16 ImageRenderer.
+  ///
+  /// - Parameters:
+  ///   - precision: The percentage of pixels that must match.
+  ///   - size: A view size override.
+  ///   - traits: A trait collection override.
+  static func imageRenderer(
+    precision: Float = 1,
+    layout: SwiftUISnapshotLayout = .sizeThatFits,
+    traits: UITraitCollection = .init()
+    )
+    -> Snapshotting {
+      let config: ViewImageConfig
+
+      switch layout {
+      #if os(iOS) || os(tvOS)
+      case let .device(config: deviceConfig):
+        config = deviceConfig
+      #endif
+      case .sizeThatFits:
+        config = .init(safeArea: .zero, size: nil, traits: traits)
+      case let .fixed(width: width, height: height):
+        let size = CGSize(width: width, height: height)
+        config = .init(safeArea: .zero, size: size, traits: traits)
+      }
+
+      let imageScale: CGFloat
+      if traits.displayScale != 0.0 {
+        imageScale = traits.displayScale
+      } else {
+        imageScale = UIScreen.main.scale
+      }
+
+      return SimplySnapshotting.image(precision: precision, scale: imageScale).asyncPullback { view in
+        return .init { callback in
+          Task { @MainActor in
```

Comment:
Is this `Task` necessary? Can the image rendered be called synchronously (and use `pullback` instead of `asyncPullback`)?